inst/include/Rcpp/platform/compiler.h

```diff
@@ -24,173 +22,35 @@
-    #if defined(__GLIBCXX__) && defined(__GXX_EXPERIMENTAL_CXX0X__)
-        #if GCC_VERSION >= 40400
-            #define HAS_CXX0X_UNORDERED_MAP
-            #define HAS_CXX0X_UNORDERED_SET
-            #define HAS_CXX0X_INITIALIZER_LIST
-        #endif
-    #endif
-#elif defined(__clang__)
-    #if __cplusplus >= 201103L
-        #if __has_include(<unordered_map>)
-            #define HAS_CXX0X_UNORDERED_MAP
-        #endif
-        #if __has_include(<unordered_set>)
-            #define HAS_CXX0X_UNORDERED_SET
-        #endif
-        #if __has_include(<initializer_list>)
-            #define HAS_CXX0X_INITIALIZER_LIST
-        #endif
-    #endif
-#endif
-
-// Check TR1 Headers
-#if defined(__INTEL_COMPILER) || (defined(__GNUC__) && !defined(__clang__))
-    #if defined(__GLIBCXX__)
-        #if GCC_VERSION >= 40400 || ( GCC_VERSION >= 40201 && defined(__APPLE__) )
-            #define HAS_TR1_UNORDERED_MAP
-            #define HAS_TR1_UNORDERED_SET
-        #endif
-    #endif
-#elif defined(__clang__)
-    #if __cplusplus >= 201103L
-        #if __has_include(<tr1/unordered_map>)
-            #define HAS_TR1_UNORDERED_MAP
-        #endif
-        #if __has_include(<tr1/unordered_set>)
-            #define HAS_TR1_UNORDERED_SET
-        #endif
-    #endif
-#endif
-
-#if defined(HAS_TR1_UNORDERED_MAP) && defined(HAS_TR1_UNORDERED_SET)
-#define HAS_TR1
-#endif
-
-// Conditionally include headers
-#ifdef HAS_CXX0X_INITIALIZER_LIST
 #include <initializer_list>
-#endif
-
-#ifdef RCPP_USING_CXX11
-    #if defined(HAS_CXX0X_UNORDERED_MAP)
-        #include <unordered_map>
-        #define RCPP_USING_UNORDERED_MAP
-        #define RCPP_UNORDERED_MAP std::unordered_map
-    #else
-        #include <map>
-        #define RCPP_USING_MAP
-        #define RCPP_UNORDERED_MAP std::map
-    #endif
-    #if defined(HAS_CXX0X_UNORDERED_SET)
-        #include <unordered_set>
-        #define RCPP_USING_UNORDERED_SET
-        #define RCPP_UNORDERED_SET std::unordered_set
-    #else
-        #include <set>
-        #define RCPP_USING_SET
-        #define RCPP_UNORDERED_SET std::set
-    #endif
-#else
-    #if defined(HAS_TR1_UNORDERED_MAP)
-        #include <tr1/unordered_map>
-        #define RCPP_USING_TR1_UNORDERED_MAP
-        #define RCPP_UNORDERED_MAP std::tr1::unordered_map
-    #else
-        #include <map>
-        #define RCPP_USING_MAP
-        #define RCPP_UNORDERED_MAP std::map
-    #endif
-    #if defined(HAS_TR1_UNORDERED_SET)
-        #include <tr1/unordered_set>
-        #define RCPP_USING_TR1_UNORDERED_SET
-        #define RCPP_UNORDERED_SET std::tr1::unordered_set
-    #else
-        #include <set>
-        #define RCPP_USING_SET
-        #define RCPP_UNORDERED_SET std::set
-    #endif
-#endif
+#include <unordered_map>
+#define RCPP_USING_UNORDERED_MAP
+#define RCPP_UNORDERED_MAP std::unordered_map
+#include <unordered_set>
+#define RCPP_USING_UNORDERED_SET
+#define RCPP_UNORDERED_SET std::unordered_set
```

Comment:
- Nobody on CRAN uses RCPP_USING_UNORDERED_MAP or RCPP_USING_UNORDERED_SET.
- Only the jiebaR package uses RCPP_UNORDERED_MAP.
- Only the fwsim package uses RCPP_UNORDERED_SET.

Reply:
Please this really is not helping. 

Look at the ten commits that got us here.  **Of course** when feasible we can remove it **but currently Rcpp itself uses it**. 

Moreover, as you seem to agree on, we always have to keep other users in mind so **many** of the things you propose here will need **individual** reverse depends run.

You cannot unbuild Rome in a day either.

Reply:
Can you reach out to those two packages?

Reply:
If not I have no issue declaring this as 'deprecated' and giving them six months or a year.  We have done that at other times.

These are the things reverse dependency checks reveal.

Reply:
I'll reach out those packages.